docs/source/error_code_list.rst

```diff
@@ -0,0 +1,502 @@
+.. _error-code-list:
+
+Error codes enabled by default
+==============================
+
+This section documents various errors codes that mypy can generate
+with default options. See :ref:`error-codes` for general documentation
+about error codes. :ref:`error-codes-optional` documents additional
+error codes that you can enable.
```

Comment:
Let's postpone the decision until after I've merged this PR, as combining the pages will make the PR commit history harder to track. 

The ordering is based on frequency in internal logs + grouping of related options. I can adjust the ordering in another PR -- again it would make the commit history confusing.